src/Compilers/CSharp/Test/Emit/Emit/CovariantReturnTests.cs

```diff
@@ -304,7 +313,74 @@ static void Main()
             var expectedOutput = @"";
             CompileAndVerify(compilation, expectedOutput: expectedOutput, verify: Verification.Skipped);
         }
-    }
+
+        [ConditionalFact(typeof(CovariantReturnRuntimeOnly))]
+        public void CheckPreserveBaseOverride_01()
+        {
+            var s0 = @"
+public class Base
+{
+    public virtual object M() => ""Base.M"";
+}
+";
+            var ref0 = CreateCovariantCompilation(
+                s0,
+                assemblyName: "ref0").VerifyEmitDiagnostics().EmitToImageReference();
+
+            var s1a = @"
+public class Mid : Base
+{
 }
+";
+            var ref1a = CreateCovariantCompilation(
+                s1a,
+                references: new[] { ref0 },
+                assemblyName: "ref1").VerifyEmitDiagnostics().EmitToImageReference();
+
+            var s1b = @"
+public class Mid : Base
+{
+    public override string M() => ""Mid.M"";
+}
+";
+            var ref1b = CreateCovariantCompilation(
+                s1b,
+                references: new[] { ref0 },
+                assemblyName: "ref1").VerifyEmitDiagnostics().EmitToImageReference();
 
-#endif
+            var s2 = @"
+public class Derived : Mid
+{
+    public override string M() => ""Derived.M"";
+}
+";
+            var ref2 = CreateCovariantCompilation(
+                s2,
+                references: new[] { ref0, ref1a },
+                assemblyName: "ref2").VerifyEmitDiagnostics().EmitToImageReference();
+
+            var program = @"
+using System;
+public class Program
+{
+    static void Main()
+    {
+        Derived d = new Derived();
+        Mid m = d;
+        Base b = m;
+        Console.WriteLine(b.M().ToString());
+        Console.WriteLine(m.M().ToString());
+        Console.WriteLine(d.M().ToString());
+    }
+}
+";
+            var compilation = CreateCovariantCompilation(program, options: TestOptions.DebugExe, references: new[] { ref0, ref1b, ref2 });
+            // compilation.VerifyDiagnostics();
```

Comment:
is comment intentional?

Reply:
No.

---
In reply to: [461750001](https://github.com/dotnet/roslyn/pull/46367#discussion_r461750001) [](ancestors = 461750001)